src/tools/sssctl/sssctl_logs.c

```diff
@@ -322,73 +515,73 @@ errno_t sssctl_debug_level(struct sss_cmdline *cmdline,
     const char *debug_as_string = NULL;
-    const char *config_file = NULL;
-    const char *pc_config_file = NULL;
+
     struct debuglevel_tool_ctx *ctx = NULL;
     struct poptOption long_options[] = {
-        {"config", 'c', POPT_ARG_STRING, &pc_config_file,
-            0, _("Specify a non-default config file"), NULL},
+        {"domain", '\0', POPT_ARG_ARGV, &pc_domains,
+            0, _("Target a specific domain"), NULL},
+        POPT_SERV_OPTION(SSSD, pc_services),
+        POPT_SERV_OPTION(NSS, pc_services),
+        POPT_SERV_OPTION(PAM, pc_services),
+        POPT_SERV_OPTION(SUDO, pc_services),
+        POPT_SERV_OPTION(AUTOFS, pc_services),
+        POPT_SERV_OPTION(SSH, pc_services),
+        POPT_SERV_OPTION(PAC, pc_services),
+        POPT_SERV_OPTION(IFP, pc_services),
         POPT_TABLEEND
     };
 
+    /* allocate context */
+    ctx = talloc_zero(NULL, struct debuglevel_tool_ctx);
+    if (ctx == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Could not allocate memory for tools context\n");
+        return ENOMEM;
+    }
+
     ret = sss_tool_popt_ex(cmdline, long_options, SSS_TOOL_OPT_OPTIONAL, NULL,
                            NULL, "DEBUG_LEVEL_TO_SET",
                            _("Specify debug level you want to set"),
-                           SSS_TOOL_OPT_REQUIRED, &debug_as_string, NULL);
+                           SSS_TOOL_OPT_OPTIONAL, &debug_as_string, NULL);
     if (ret != EOK) {
         DEBUG(SSSDBG_CRIT_FAILURE, "Unable to parse command arguments\n");
-        return ret;
-    }
-
-    /* get config file */
-    if (pc_config_file) {
-        config_file = talloc_strdup(ctx, pc_config_file);
-    } else {
-        config_file = talloc_strdup(ctx, SSSD_CONFIG_FILE);
-    }
-
-    if (config_file == NULL) {
-        DEBUG(SSSDBG_CRIT_FAILURE, "talloc_strdup() failed\n");
-        ret = ENOMEM;
         goto fini;
     }
 
     CHECK_ROOT(ret, debug_prg_name);
 
-    debug_to_set = parse_debug_level(debug_as_string);
-    CHECK(debug_to_set == SSSDBG_INVALID, fini, "Invalid debug level.");
-
-    /* allocate context */
-    ctx = talloc_zero(NULL, struct debuglevel_tool_ctx);
-    if (ctx == NULL) {
-        DEBUG(SSSDBG_CRIT_FAILURE,
-              "Could not allocate memory for tools context\n");
-        ret = ENOMEM;
-        goto fini;
+    if (debug_as_string != NULL) {
+        debug_to_set = (uint32_t) parse_debug_level(debug_as_string);
+        CHECK(debug_to_set == SSSDBG_INVALID, fini, "Invalid debug level.");
     }
 
+    /* Create a list with all the target names (services + domains) */
+    targets = get_targets(ctx, pc_services, pc_domains);
+    CHECK(targets == NULL, fini, "Could not allocate memory.");
+
     ret = connect_to_confdb(ctx, &ctx->confdb);
     CHECK(ret != EOK, fini, "Could not connect to configuration database.");
 
     ret = get_confdb_sections(ctx, ctx->confdb, &ctx->sections);
     CHECK(ret != EOK, fini, "Could not get all configuration sections.");
```

Comment:
It seems this (and entire `debuglevel_tool_ctx`) isn't used anymore (and thus could be removed)?